Germany

joos2010kj opened the pull request:
- Add new params: run name, unique, and project name
    - Fix the germany and south africa's S1S2 dataset indexing issue
    - Run model with germany dataset (baseline)
    - Modify commands for Germany submission

joos2010kj committed:
[temp: todo] update suffix in helper.py

joos2010kj committed:
[WIP] Fix the indexing error

Added germany to the params list.

Need to look into line 68 in sentinel_1_reader.py when it crashes w/ iloc but does not crash with loc.  Running:

python train.py --competition germany --satellite sentinel_1 --pos 33N_18E_242N

NOTE: It requires medium gpu due to RAM issue for s1_s2 operation.

joos2010kj committed:
Merge branch 'planet-daily-data' of github.com:ivanzvonkov/tum-planet-radearth-ai4food-challenge into planet-daily-data

joos2010kj committed:
Merge branch 'hyekang' into planet-daily-data

Conflicts:
	helper.py
	src/utils/data_loader.py
	train.py

joos2010kj committed:
Add new params: run name, unique, and project name

joos2010kj committed:
Fix the Germany & South Africa's S1S2 dataset indexing issue

joos2010kj committed:
Run model with germany dataset (baseline)

joos2010kj committed:
Modify commands for submission

joos2010kj committed:
Merge branch 'germany' into main

Conflicts:
	helper.py
	src/utils/s1_s2_reader.py
	train.py

joos2010kj commented on train.py:
```diff
@@ -143,7 +173,7 @@
 config["clip_value"] = clip_value
 for p in model.parameters():
-    if p.requires_grad:
+    if p.requires_grad and config['competition'] == 'south_africa':
```
loss is nan if this condition is in place for germany

ivanzvonkov commented on src/utils/s1_s2_reader.py:
```diff
@@ -71,9 +71,7 @@ def __init__(
-        assert self.s1_reader.labels.drop("path", axis=1).equals(
-            self.s2_reader.labels.drop("path", axis=1)
-        )
+        # assert self.s1_reader.labels.equals(self.s2_reader.labels)
```
Why the comment?

joos2010kj replied:
Forgot.  Removed.

ivanzvonkov commented on submit.py:
```diff
@@ -98,10 +104,11 @@
 submission_path.parent.mkdir(parents=True, exist_ok=True)
 output_frame.to_json(submission_path)
 
+import ipdb; ipdb.set_trace()
```
I think this line will fail inside sbatch

joos2010kj replied:
Right, forgot.  Removed.

ivanzvonkov commented on train.py:
```diff
@@ -55,11 +62,15 @@
-arg_parser.add_argument(
-    "--alignment", type=str, default="1to2", help="Can be: 1to2 or 2to1 (76 vs. 41)"
-)
+arg_parser.add_argument("--name", type=str, default=None, help="Manually the run name (e.g., snowy-owl-10); None for automatic naming.")
```
What's the use case for naming runs?

joos2010kj replied:
It's been pretty helpful for me in identifying the run in wandb sweep.  Yes, you can rename it on the website, but I found naming it before sending it off more convenient.

ivanzvonkov replied:
what to you mean identifying? What do you name them?

ivanzvonkov replied:
Ok it keeps the same behavior I am okay with this

ivanzvonkov commented on train.py:
```diff
@@ -26,9 +28,14 @@
+    type=str, 
+    default="germany",
+    help="germany, south_africa"
+)
```
If the original wandb project is for south africa then it would make sense for the default competition to be south africa too probably

ivanzvonkov commented on train.py:
```diff
@@ -143,7 +173,7 @@
 config["clip_value"] = clip_value
 for p in model.parameters():
-    if p.requires_grad:
+    if p.requires_grad and config['competition'] == 'south_africa':
```
What's going on here?

ivanzvonkov replied:
> this clipping is needed for Germany dataset

Right now this code disables clipping for the Germany dataset, unless I am misunderstanding something

> loss is nan

Do you know why this is?

ivanzvonkov replied:
I think this is the only thing that needs to be resolved!

ivanzvonkov replied:
Any update on this?

joos2010kj replied:
I've been working on getting the LSTM implementation first, so I'll get back to it once it's over.

ivanzvonkov commented on train.py:
```diff
@@ -162,7 +192,8 @@
         entity="nasa-harvest",
-        project="ai4food-challenge",
+        project=config['project'],
+        name=config['name'],
```
Is this optional? Ie will it work if not specified?

joos2010kj replied:
Yes, the new params I added (--name, --unique, --project) all work as how it did before if left unspecified.  In other words, it will assign a random name and send it off to your wandb storage, not mine.

ivanzvonkov commented on train.py:
```diff
@@ -91,11 +118,13 @@
     label_names_259, reader_259 = load_reader(pos="34S_19E_259N", **kwargs)
     print("\u2713 Loaded 259")
+    reader_258.labels.reset_index(inplace=True)
+    reader_259.labels.reset_index(inplace=True)
```
1. Why not reset the index after concatenation?
2. Why does it work for South Africa data as is, but this change is needed for the Germany data?

joos2010kj replied:
1. As mentioned in slack, I have to place the reset_index at those three places for fixing the iloc / loc issue.  And no, just putting one reset_index at the end of the whole if-else block doesn’t work because of the nature of ConcatDataset — it references reader_258 and reader_259 directly (those are not touched other than for accessing elements) and ConcatDataset keeps a different index series than those two.

More specifically, when you print ConcatDataset, you will see that its whole dataset is indexed sequentially (i.e., len = index), even if their subset datasets were initially not indexed sequentially (i.e., len < index).  Thus, resetting the index for the concatenated dataset has no impact.  However, if you apply reset_index to those two subsets BEFORE concatenation, then it's good.  Unless the index is the same as FID, reset_index will not have an issue.

joos2010kj replied:
2. < Give me a second >

ivanzvonkov replied:
This makes sense

joos2010kj reviewed (commented):
Updated in the following commit and answered comments

joos2010kj committed:
Fix loss nan, cuda issues, spatiotemporal combo incompatibility issues in Germany dataset

1. Remove comments and ipdb
2. Add competition-to-pos condition
3. Add wandb command
4. Fix loss nan: train_valid_eval_utils.py giving inconsistent output
- a. Loss Nan was addressed by filtering out all npz files containing np.inf or np.nan from S1_Reader and S2_Reader
- ** np.inf and/or np.nan were present only in S1_germany, and not in south_africa or other germany datasets.
- ** no recognizable pattern found among np.inf and np.nan, so decided to filter them out.
5. cuda issue & allowing every combination
- a. the size of mlp4 (src/utils/ltae - mlp4) was fixed.
- ** The last channel of mlp4 should be equivalent to the number of classes, but it was fixed to 5 (south africa dataset).

ivanzvonkov commented:
Okay, looks good. Please try the npz delete we talked about, that may eliminate the need for the filter!

ivanzvonkov commented:
Is this good to merge?

joos2010kj commented:
~Let me take a look at the merge conflicts now and you review the changes~

joos2010kj commented on helper.py:
```diff
@@ -55,6 +55,13 @@ def load_reader(
 
     label_file = f"{root}/{country}_{train_or_test}_labels/{country}_{train_or_test}_labels_{pos}/labels.geojson"
     labels = gpd.read_file(label_file)
+
```
Needed for nan/inf value filtering from S1, S1_S2, and S1_S2_PlanetFusion

joos2010kj commented on src/s1g_redflag.txt:
```diff
@@ -0,0 +1,85 @@
+52644
```
List of fids in germany sentinel 1 dataset containing inf/nan.  No access to test set, so can't remove them from test set.

joos2010kj commented on train.py:
```diff
@@ -181,7 +183,7 @@
 config["clip_value"] = clip_value
 for p in model.parameters():
-    if p.requires_grad and config['competition'] == 'south_africa':
+    if p.requires_grad:
```
Reverted back to original

joos2010kj reviewed (commented):
Just a few comments in case difficult to follow.  ~LMK if it looks good one last time and I'll merge.~

joos2010kj commented:
I will have to review codes again since it's crashing after merging when I run the code.

ivanzvonkov commented:
Let me know if you need a second look

joos2010kj committed:
Merge branch 'main' into working-clone

Conflicts:
	helper.py
	src/utils/s1_s2_planet_reader.py
	src/utils/s1_s2_reader.py
	src/utils/sentinel_1_reader.py
	src/utils/sentinel_2_reader.py
	train.py
